lib/src/game/base_game.dart

```diff
@@ -125,6 +129,15 @@ class BaseGame extends Game with FPSCounter {
       c.onRemove();
       components.remove(c);
     });
+
+    if (this is HasCollidables) {
+      final collidables = (this as HasCollidables).collidables;
```

Comment:
Is this casting really necessary? Since this is in an `if` which already makes sure that this class is `HasCollidables`, this cast shouldn't be necessary

Reply:
I think he commented there is no smart cast for `this`

Reply:
an oversight of dart of course, this is effectively immutable